Agent unable to set ownerReference due to missing RBAC privileges

https://devops-ci.elastic.co/job/cloud-on-k8s-e2e-tests-ocp/340/testReport/github/com_elastic_cloud-on-k8s_test_e2e_agent/TestSystemIntegrationConfig_Agent_status_should_be_updated/

fails with 
```
=== RUN   TestSystemIntegrationConfig/Agent_status_should_be_updated
Retries (30m0s timeout): .......................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................
    utils.go:84: 
        	Error Trace:	utils.go:84
        	Error:      	Received unexpected error:
        	            	expected status {Version:7.10.1 ExpectedNodes:0 AvailableNodes:0 Health:green ElasticsearchAssociationsStatus:} but got {Version: ExpectedNodes:0 AvailableNodes:0 Health: ElasticsearchAssociationsStatus:}
        	Test:       	TestSystemIntegrationConfig/Agent_status_should_be_updated
{"log.level":"error","@timestamp":"2021-01-12T02:06:14.812Z","message":"stopping early","service.version":"0.0.0-SNAPSHOT+00000000","service.type":"eck","ecs.version":"1.4.0","error":"test failure","error.stack_trace":"github.com/elastic/cloud-on-k8s/test/e2e/test.StepList.RunSequential\n\t/go/src/github.com/elastic/cloud-on-k8s/test/e2e/test/step.go:43\ngithub.com/elastic/cloud-on-k8s/test/e2e/agent.TestSystemIntegrationConfig\n\t/go/src/github.com/elastic/cloud-on-k8s/test/e2e/agent/config_test.go:46\ntesting.tRunner\n\t/usr/local/go/src/testing/testing.go:1123"}
    --- FAIL: TestSystemIntegrationConfig/Agent_status_should_be_updated (1800.00s)
```

My assumption is that this error message from the logs might be relevant: 

```
{"log.level":"error","@timestamp":"2021-01-12T01:52:30.616Z","log.logger":"controller","message":"Reconciler error","service.version":"1.5.0-SNAPSHOT+f9244f6a","service.type":"eck","ecs.version":"1.4.0","controller":"agent-es-association-controller","name":"test-agent-system-int-84gf","namespace":"e2e-e2buv-mercury","error":"secrets \"test-agent-system-int-84gf-e2e-e2buv-mercury-test-agent-system-int-r4xx-agent-user\" is forbidden: cannot set blockOwnerDeletion if an ownerReference refers to a resource you can't set finalizers on: , <nil>","errorCauses":[{"error":"secrets \"test-agent-system-int-84gf-e2e-e2buv-mercury-test-agent-system-int-r4xx-agent-user\" is forbidden: cannot set blockOwnerDeletion if an ownerReference refers to a resource you can't set finalizers on: , <nil>"}],"error.stack_trace":"sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.6.3/pkg/internal/controller/controller.go:246\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.6.3/pkg/internal/controller/controller.go:218\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).worker\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.6.3/pkg/internal/controller/controller.go:197\nk8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1\n\t/go/pkg/mod/k8s.io/apimachinery@v0.18.14/pkg/util/wait/wait.go:155\nk8s.io/apimachinery/pkg/util/wait.BackoffUntil\n\t/go/pkg/mod/k8s.io/apimachinery@v0.18.14/pkg/util/wait/wait.go:156\nk8s.io/apimachinery/pkg/util/wait.JitterUntil\n\t/go/pkg/mod/k8s.io/apimachinery@v0.18.14/pkg/util/wait/wait.go:133\nk8s.io/apimachinery/pkg/util/wait.Until\n\t/go/pkg/mod/k8s.io/apimachinery@v0.18.14/pkg/util/wait/wait.go:90"}
```

I believe we removed the finalizer privilege because we thought we would not need it. What seems odd is that tests fail not more consistently.

--- a/config/e2e/monitoring.yaml
+++ b/config/e2e/monitoring.yaml
@@ -612,6 +612,7 @@ rules:
   resources:
   - agents
   - agents/status
+  - agents/finalizers
   verbs:
   - get
   - list

--- a/config/e2e/operator.yaml
+++ b/config/e2e/operator.yaml
@@ -149,6 +149,7 @@ rules:
   resources:
   - agents
   - agents/status
+  - agents/finalizers
   verbs:
   - get
   - list

--- a/deploy/eck-operator/templates/_helpers.tpl
+++ b/deploy/eck-operator/templates/_helpers.tpl
@@ -210,7 +210,7 @@ RBAC permissions
   resources:
   - elasticsearches
   - elasticsearches/status
-  - elasticsearches/finalizers
+  - elasticsearches/finalizers # needed for ownerReferences with blockOwnerDeletion on OCP
   - enterpriselicenses
   - enterpriselicenses/status
   verbs:
@@ -226,7 +226,7 @@ RBAC permissions
   resources:
   - kibanas
   - kibanas/status
-  - kibanas/finalizers
+  - kibanas/finalizers # needed for ownerReferences with blockOwnerDeletion on OCP
   verbs:
   - get
   - list
@@ -240,7 +240,7 @@ RBAC permissions
   resources:
   - apmservers
   - apmservers/status
-  - apmservers/finalizers
+  - apmservers/finalizers # needed for ownerReferences with blockOwnerDeletion on OCP
   verbs:
   - get
   - list
@@ -254,7 +254,7 @@ RBAC permissions
   resources:
   - enterprisesearches
   - enterprisesearches/status
-  - enterprisesearches/finalizers
+  - enterprisesearches/finalizers # needed for ownerReferences with blockOwnerDeletion on OCP
   verbs:
   - get
   - list
@@ -268,7 +268,7 @@ RBAC permissions
   resources:
   - beats
   - beats/status
-  - beats/finalizers
+  - beats/finalizers # needed for ownerReferences with blockOwnerDeletion on OCP
   verbs:
   - get
   - list
@@ -282,6 +282,7 @@ RBAC permissions
   resources:
   - agents
   - agents/status
+  - agents/finalizers # needed for ownerReferences with blockOwnerDeletion on OCP
   verbs:
   - get
   - list

